### PR TITLE
device init IP checks all interfaces

### PR DESCRIPTION
**Pull request by halfway258**

### CODE REVIEW

File: src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl [hunk 1]
```diff
@@ -313,4 +313,41 @@ getHostName() ->
...
+    ok.
+
+% get this host ip
+getdeviceIP() ->
+    {ok, IFList} = inet:getifaddrs(),    % IFList format: [{IF_name, [{field, param},{},...]},...]
+    SubnetsList = getNerlSubnets(),
+    getdeviceIP(IFList, SubnetsList).
+
+getdeviceIP([], SubnetsList) ->
+    logger:error("No supported interface was found. Current supported interfaces list is: ~p",[SubnetsList]);
```
Comment (leondavi) [inline src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl:325]: Add to error log an explanation to edit the file: NerlNet_subnets_config

File: NerlNet_subnets_config [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# List of relevant subnets that NerlNet will use. Instead of 192.168.x.x write 192.168.
+
+192.168.
+10.0.
+192.168.100.
```
Comment (leondavi) [inline NerlNet_subnets_config:5]: Remove 192.168.100
It was only for example 🙂

**Review by leondavi — CHANGES_REQUESTED** [review]

File: src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl [hunk 3]
```diff
@@ -46,7 +46,7 @@
 
 
 start(_StartType, _StartArgs) ->
-     HostName = getHostName(),
+     HostName = getdeviceIP(),
      %HostName = "127.0.0.1",        %TODO: update jsons with real ips
      io:format("My HostName: ~p~n",[HostName]),
```
Comment (leondavi) [inline src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl:51]: Change io:format to logger:notice("This device IP: ~p~n", [DeviceIP]")

File: src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl [hunk 4]
```diff
@@ -46,7 +46,7 @@
 
 
 start(_StartType, _StartArgs) ->
-     HostName = getHostName(),
+     HostName = getdeviceIP(),
```
Comment (leondavi) [inline src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl:49]: Change HostName to DeviceIP

**Review by leondavi — CHANGES_REQUESTED** [review]

File: src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl (hunk 1 above)
Comment (halfway258) [reply src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl]: done

File: NerlNet_subnets_config (hunk 2 above)
Comment (halfway258) [reply NerlNet_subnets_config]: done

File: src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl (hunk 3 above)
Comment (halfway258) [reply src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl]: done

File: src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl (hunk 4 above)
Comment (halfway258) [reply src_erl/Communication_Layer/http_Nerlserver/src/nerlNetServer_app.erl:49]: done

**Review by leondavi — APPROVED** [review]

**Pull request merged**